tensorboard/components/tf_logging/logging.ts

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+/*
+ * Utilities for logging.
+ */
+
+
+/**
+ * An interface for events logged to Analytics. See
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/events
+ */
+export interface AnalyticsEvent {
+  // This should usually be 'event'.
+  hitType: string;
+
+  // This should be the relevant plugin name.
+  eventCategory: string;
+
+  // This should be a string describing the action taken.
+  eventAction: string;
+
+  // This string label contains more information.
+  eventLabel: string;
```

Comment:
Do you think that we should follow a convention that this is always a JSON object? This would give us some forward/backward compatibility if we change event types.

For instance, you currently send `('event', 'core', 'dashboardSelect', 'images')`. But maybe we want to change that to `('event', 'core', 'dashboardSelect', '{"old": "scalars", "new": "images"}')`. If the old form were `{"new": "images"}` then old events would be compatible with the new format.

(Ideally, we'd use protos, but I suspect that integrating them is not worth the effort.)

Reply:
Reverted to faking `ga` and recording a page view.